transport/http.go

```diff
@@ -116,26 +147,41 @@ func NewHTTPTransport() (*HTTPTransport, error) {
 	if err != nil {
 		return nil, err
 	}
-
 	serverTimeout, err := configutil.ParseDurationEnv(envServerTimeout, defaultServerTimeout)
 	if err != nil {
 		return nil, err
 	}
 	if serverTimeout < 0 {
 		serverTimeout = 0
 	}
-
 	serverURLs, err := initServerURLs()
 	if err != nil {
 		return nil, err
 	}
 
-	tlsConfig := &tls.Config{InsecureSkipVerify: !verifyServerCert}
-	serverCertPath := os.Getenv(envServerCert)
-	if serverCertPath != "" {
-		serverCert, err := loadCertificate(serverCertPath)
+	opts := HTTPTransportOptions{
+		SkipServerVerify: !verifyServerCert,
+		ServerURLs:       serverURLs,
+		ServerTimeout:    serverTimeout,
+		ServerTLSCert:    os.Getenv(envServerCert),
+		ServerCACert:     os.Getenv(envCACert),
+	}
+	if apiKey := os.Getenv(envAPIKey); apiKey != "" {
+		opts.APIKey = apiKey
+	} else if secretToken := os.Getenv(envSecretToken); secretToken != "" {
+		opts.SecretToken = secretToken
+	}
+	return NewHTTPTransportOptions(opts)
+}
+
+// NewHTTPTransportOptions returns a customized HTTPTransport which can be used
+// for streaming data to the APM Server.
+func NewHTTPTransportOptions(opts HTTPTransportOptions) (*HTTPTransport, error) {
```

Comment:
How about `NewHttpTransportFromOptions`? When reading `NewHTTPTransportOptions` I'd expect a `HTTPTransportOptions` struct to be returned.

Reply:
For better or worse, the naming is consistent with other existing functions like `NewTracerOptions`. I'd prefer to keep them consistent even if it's not the most clear language.